master/internal/rm/kubernetesrm/pods.go

```diff
@@ -334,6 +364,185 @@ func (p *pods) getSystemResourceRequests(ctx *actor.Context) error {
 	return nil
 }
 
+func (p *pods) reattachAllocationPods(ctx *actor.Context, msg reattachAllocationPods) error {
+	listOptions := metaV1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", determinedLabel, msg.allocationID),
+	}
+
+	pods, err := p.podInterface.List(context.TODO(), listOptions)
+	if err != nil {
+		return errors.Wrap(err, "error listing pods checking if they can be restored")
+	}
+
+	configMaps, err := p.configMapInterface.List(context.TODO(), listOptions)
+	if err != nil {
+		return errors.Wrap(err, "error listing config maps checking if they can be restored")
+	}
+	existingConfigMaps := make(map[string]bool)
+	for _, cm := range configMaps.Items {
+		existingConfigMaps[cm.Name] = true
+	}
+
+	var containerIDs []string
+	var k8sPods []*k8sV1.Pod
+	for _, pod := range pods.Items {
+		found := false
+		for _, container := range pod.Spec.Containers {
+			for _, env := range container.Env {
+				if env.Name == "DET_CONTAINER_ID" {
+					if !existingConfigMaps[pod.Name] {
+						p.deleteAllocationResources(ctx, pods, configMaps)
+						ctx.Respond(fmt.Errorf("pod missing config map %s", pod.Name))
+						return nil
+					}
+
+					p := pod
+					k8sPods = append(k8sPods, &p)
+					containerIDs = append(containerIDs, env.Value)
+
+					found = true
+					break
+				}
+			}
+			if found {
+				break
+			}
+		}
+	}
+
+	if len(k8sPods) != msg.numPods {
+		p.deleteAllocationResources(ctx, pods, configMaps)
+		ctx.Respond(fmt.Errorf("not enough pods found for allocation"))
+		return nil
```

Comment:
i think there are some cases here where it would be OK to restore, depending on how the pods exited and which pods they are.

Reply:
Talked offline, decided it would be hard to differentiate the case where it would be OK to restore

An example case that was discussed was a dtrain job where workers exited and were basically done (but we lost track of them on restore) and the chief was still reporting metrics. In this case it could be possible to continue but overall somewhat tough to identify cases like this.